### PR TITLE
Unit Tests & Documentation

### PR DESCRIPTION
**Pull request by LucaHaverty**
### Description
Unit tests for the major systems that I worked on this summer

### Objectives
- [x] Input System Tests
- [x] Preferences System Tests
- [x] Select Menu Tests
- [x] Lots of documentation

### CODE REVIEW

**Review by azaleacolburn — APPROVED** [review]
Super Cool!

File: simulation/samples/JavaSample/.settings/org.eclipse.buildship.core.prefs
Comment (BrandonPacewic) [inline simulation/samples/JavaSample/.settings/org.eclipse.buildship.core.prefs:1]: Does this file need to be tracked? Seems like it updates with a specific user path every time we run codesim.

Comment (LucaHaverty) [reply simulation/samples/JavaSample/.settings/org.eclipse.buildship.core.prefs:1]: Nope. I reverted those changes

File: fission/src/ui/modals/configuring/SettingsModal.tsx [hunk 1]
```diff
@@ -66,7 +66,7 @@ const SettingsModal: React.FC<ModalPropsImpl> = ({ modalId }) => {
                 saveSettings()
             }}
         >
-            <div className="flex overflow-y-auto flex-col gap-2 bg-background-secondary rounded-md p-2 min-w-[22vw]">
+            <div className="flex overflow-y-auto flex-col gap-2 bg-background-secondary rounded-md p-2 max-h-[60vh]">
```
Comment (BrandonPacewic) [inline fission/src/ui/modals/configuring/SettingsModal.tsx]: This looks like the same change from #1100.

Comment (LucaHaverty) [reply fission/src/ui/modals/configuring/SettingsModal.tsx]: @BrandonPacewic I probably thought that I broke it in this PR then forgot about that until I did the hotfix. It's such a small fix so should I just close the hotfix pr?

Comment (BrandonPacewic) [reply fission/src/ui/modals/configuring/SettingsModal.tsx]: Either way, I would probably just revert the line change in this PR. Makes it more explicit for people to test but it's up to you.

**Review by BrandonPacewic — APPROVED** [review]
Love it 🪺

**Review by Dhruv-0-Arora — APPROVED** [review]
Splendid!

**Pull request merged**